### PR TITLE
Add UV attribute to click event.

### PR DESCRIPTION
**Pull request by bnolan**
**Description:**

Add uv attribute to detail in click event.

**Changes proposed:**
- Add uv attribute to detail in click event.
- Don't propagate uv attribute up to parent elements

### CODE REVIEW

File: examples/test-cursor/index.html [hunk 1]
```diff
@@ -61,7 +61,8 @@
         var clickedEl = null;
         var els = document.querySelectorAll('a-entity');
         Array.prototype.forEach.call(els, function (el) {
-          el.addEventListener('click', function () {
+          el.addEventListener('click', function (e) {
+            console.log('uv', e.detail.uv);
```
Comment (dmarcos) [inline examples/test-cursor/index.html]: Remove log

File: src/components/cursor.js [hunk 2]
```diff
@@ -50,8 +50,13 @@ module.exports.Component = registerComponent('cursor', {
 
   emit: function (evt) {
     var intersectedEl = this.intersectedEl;
-    this.el.emit(evt, { target: this.intersectedEl });
-    if (intersectedEl) { intersectedEl.emit(evt); }
+    var dispatchEvent = Object.assign({}, this.intersectionEvent, {
```
Comment (dmarcos) [inline src/components/cursor.js]: Why do we need to copy this.intersectionEvent?

File: src/components/cursor.js [hunk 3]
```diff
@@ -65,6 +70,9 @@ module.exports.Component = registerComponent('cursor', {
     var data = this.data;
     var el = evt.detail.el;
     var distance = evt.detail.distance;
+    this.intersectionEvent = {
```
Comment (dmarcos) [inline src/components/cursor.js]: Keep one liner

**Comment by dmarcos** [comment]
Thanks! A few changes + squash and we ship this :)

File: src/components/raycaster.js [hunk 4]
```diff
@@ -85,9 +85,9 @@ module.exports.Component = registerComponent('raycaster', {
   /**
    * Remembers the last intersected element
    */
-  setExistingIntersection: function (el, distance) {
+  setExistingIntersection: function (el, distance, uv) {
     this.intersectedEl = el;
-    this.el.emit('intersection', { el: el, distance: distance });
+    this.el.emit('intersection', { el: el, distance: distance, uv: uv });
```
Comment (ngokevin) [inline src/components/raycaster.js]: I would just get rid of distance/uv and return the whole intersection object.

File: src/components/cursor.js (hunk 2 above)
Comment (cvan) [inline src/components/cursor.js]: we have an npm package that polyfills `Object.assign` where it's not natively supported. to use it, call `utils.extend`.

**Comment by ngokevin** [comment]
I plan on handling everything at https://github.com/aframevr/aframe/pull/1196, such as returning full intersection data. Thanks for the PR though! Sorry for your time.

**Pull request closed without merging**